linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1251,8 +1251,9 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
         case ArrayLength(array)      => test(array)
         case RecordSelect(record, _) => test(record)
         case IsInstanceOf(expr, _)   => test(expr)
-        case IdentityHashCode(expr)  => test(expr)
         case GetClass(arg)           => test(arg) // may NPE but that is UB.
+        case Clone(arg)              => test(arg) // may NPE but that is UB.
```

Comment:
Hum, maybe I'm not understanding the definition of "pureness" correctly here: Won't `Clone` return a different result every time it is called? Or in other words, if `Clone` is pure, why is `NewArray` not?

Reply:
Ah, well, you're understanding perfectly well that I made a basic mistake ^^ It is side-effect-free but it is not pure! I'll fix it.

Reply:
Maybe we should document what we mean by "pure" in `isPureExpression` :) (IIUC, referentially transparent n.b. with object identity and assignment).